snuba/settings/validation.py

```diff
@@ -86,12 +90,22 @@ def validate_settings(locals: Mapping[str, Any]) -> None:
                 # that are not defined in StorageSetKey.
                 pass
 
-    for logical_part in range(0, SENTRY_LOGICAL_PARTITIONS):
-        physical_part = locals["LOGICAL_PARTITION_MAPPING"].get(str(logical_part))
-        partition_count = locals["LOCAL_SLICES"]
-        assert (
-            physical_part is not None
-        ), f"missing physical slice for logical partition {logical_part}"
-        assert (
-            physical_part < locals["LOCAL_SLICES"]
-        ), f"physical slice for logical partition {logical_part} is {physical_part}, but only {partition_count} physical slices exist"
+    for storage in locals["LOGICAL_PARTITION_MAPPING"]:
+        storage_mapping = locals["LOGICAL_PARTITION_MAPPING"][storage]
+
+        # check if this is a sliced storage
+        if storage in locals["SLICED_STORAGES"]:
+            defined_slice_count = locals["SLICED_STORAGES"][storage]
+        else:
+            defined_slice_count = 1
```

Comment:
I think it makes more sense to invert this a little and make the outer loop

```python
for storage in locals["SLICED_STORAGES"]:
   assert storage in locals["LOGICAL_PARTITION_MAPPING"], "sliced mapping must be defined for sliced storage {storage}"
   # validate all logical partitions are assigned to a reasonable slice
```

does that make sense or is there a reason it's structured this way?

Reply:
Oh, right. So this was written originally under the assumption that all storages will be defined in `LOGICAL_PARTITION_MAPPING`, whether sliced or not. However, if I understand correctly, it is now such that both `LOGICAL_PARTITION_MAPPING` and `SLICED_STORAGES` are only defined for sliced storages. In this case, I'm not sure we can have validation for the non-sliced storages. They won't be in `LOGICAL_PARTITION_MAPPING`, so we can't check if they have valid logical:physical mappings.

Reply:
yeah, I wouldn't bother worrying about the case where `LOGICAL_PARTITION_MAPPING` doesn't contain a storage key (assuming `SLICED_STORAGES` also does not contain that storage)

Reply:
Ok, this makes sense. The inversion above also makes sense since `SLICED_STORAGES` is where we note down which storages are sliced. I'll make that change.